packages/url_launcher/url_launcher_platform_interface/lib/method_channel_url_launcher.dart

```diff
@@ -14,7 +13,7 @@ const MethodChannel _channel = MethodChannel('plugins.flutter.io/url_launcher');
 /// An implementation of [UrlLauncherPlatform] that uses method channels.
 class MethodChannelUrlLauncher extends UrlLauncherPlatform {
   @override
-  Future<bool> canLaunch(String url) {
+  Future<bool?> canLaunch(String url) {
```

Comment:
Returning a nullabe bool seems odd. Should it just be bool?

Reply:
I also felt the same, but since the documentation of the `invokeMethod` explicitly mentioned the return value could be `null` I decided for allowing `null`.

I could of course check for `null` on the Dart side and return `false` if this is the case. Would this be the preferred solution? Enforcing non-null using the `!` operator also doesn't feel correct since we have no real control on what's happening on the native side (or even inside the method channel logic).

Reply:
I think it is fair to do a check in `invokeMethod` and return `false` if the return value is `null`.

However, if we do that, it might counts as a breaking change. @blasten Just wanted to make sure, we are going to introduce a major bump regardless of a public api breaking change or not, right?

Reply:
Just to make sure, another consequence would be I'd have to await the future inside the `canLaunch` and `launch` methods to check the value. For example:

```dart
  @override
  Future<bool> canLaunch(String url) async {
    final ableToLaunch = await _channel.invokeMethod<bool>(
      'canLaunch',
      <String, Object>{'url': url},
    );

    return ableToLaunch ?? false;
  }
```

Reply:
Thanks! Yeah, I have confirmed with @blasten offline, all the NNBD migrations are going to be a major bump, so making breaking changes in public API (like this one) is harmless.

Reply:
Does this mean this PR can be approved?

Reply:
> all the NNBD migrations are going to be a major bump, so making breaking changes in public API (like this one) is harmless

This is the general guidance, and the default option.  There are cases where doing a major bump is less desired.

Reply:
I think we reached a general consensus that we can leave the major bump, so I will leave it as is (see also feedback from Amir on [CHANGELOG.md](https://github.com/flutter/plugins/pull/3142#discussion_r504294180))